### DISCUSSION TITLE
creating collections from same class

### DISCUSSION THREAD

**Discussion by aster2709** [typegoose/typegoose#668] [Q&A] [answered]
- Hi team. I'm wanting to have multiple collections which are going to have the same schema. so I tried this way
```ts
import { getModelForClass, prop } from "@typegoose/typegoose"
import mongoose from "mongoose"

const connectDB = async () => {
  await mongoose.connect("mongodb://localhost:27017/typegooseDB")
}

connectDB()

class Car {
  @prop()
  public color!: string
}

const CarModelA = getModelForClass(Car, { schemaOptions: { collection: "modelA" } })
const CarModelB = getModelForClass(Car, { schemaOptions: { collection: "modelB" } })

const create = async () => {
  await CarModelA.create({ color: "red" })
  await CarModelB.create({ color: "blue" })
}

create()
```
- after running these I check mongo
```sh
> show collections
modelA
> db.modelA.find().pretty()
{ "_id" : ObjectId("61ff93762f28003b454dd73e"), "color" : "red", "__v" : 0 }
{
        "_id" : ObjectId("61ff93762f28003b454dd741"),
        "color" : "blue",
        "__v" : 0
}
>
```
- as you see, it only created `modelA` and put both documents inside it.
- if this is not the correct way, please guide me how I can achieve this! thanks
- env
  - linux ubuntu `20.04` LTS
  - node `14.17.1`
  - typegoose `^9.6.0` mongoose `^6.2.0`

**Comment by hasezoey** [comment, marked as answer]
if you want a single class to have over multiple collections, you will have to change the name of the model to be used, otherwise mongoose will only create one of the models (or overwrite the older one, i dont know how it will react)

there are multiple ways to go about this:
- create a model on multiple connections
- change the name on each model

the easiest way would be to use a different name for each model, which in typegoose can be easily done by activating [`automaticName: true`](https://typegoose.github.io/typegoose/docs/api/decorators/model-options#automaticname)
and read through [Typegoose's Name Generation](https://typegoose.github.io/typegoose/docs/guides/advanced/name-generation) to see how typegoose generates names

**Comment by aster2709** [comment]
`automaticName: true` solved it. thank you very much!

**Comment by Inoir** [comment]
Just to give another way: i got it working like this:

```ts
export function getRepository(name: string) {
  return getModelForClass(CLASSMODEL, {
    schemaOptions: {
      collection: name,
    },
    options: {
      customName: () => name,
    },
  });
}
```

so you've full control over the collection name
dunno if this is intentional, but works

**Reply by hasezoey** [reply]
>  dunno if this is intentional, but works

this is intentional
see:
- #502 which made it possible to set the collection name from `getModelForClass` (not just via `@modelOptions`), because of internal caching with the name
- #469 which proposed that `customName` to be a deferred function
